server/src/main/java/com/thoughtworks/go/config/update/CreateConfigRepoCommand.java

```diff
@@ -20,20 +20,24 @@
 import com.thoughtworks.go.config.remote.ConfigRepoConfig;
 import com.thoughtworks.go.plugin.access.configrepo.ConfigRepoExtension;
 import com.thoughtworks.go.server.domain.Username;
+import com.thoughtworks.go.server.materials.MaterialUpdateService;
+import com.thoughtworks.go.server.service.MaterialConfigConverter;
 import com.thoughtworks.go.server.service.SecurityService;
 import com.thoughtworks.go.server.service.result.HttpLocalizedOperationResult;
 
 public class CreateConfigRepoCommand extends ConfigRepoCommand {
     private final ConfigRepoConfig configRepo;
 
     public CreateConfigRepoCommand(SecurityService securityService, ConfigRepoConfig configRepo, Username username,
-                                   HttpLocalizedOperationResult result, ConfigRepoExtension configRepoExtension) {
-        super(securityService, configRepo, username, result, configRepoExtension);
+                                   HttpLocalizedOperationResult result, ConfigRepoExtension configRepoExtension,
+                                   MaterialUpdateService materialUpdateService, MaterialConfigConverter converter) {
+        super(securityService, configRepo, username, result, configRepoExtension, materialUpdateService, converter);
         this.configRepo = configRepo;
     }
 
     @Override
     public void update(CruiseConfig preprocessedConfig) {
         preprocessedConfig.getConfigRepos().add(configRepo);
+        scheduleMaterialUpdate(configRepo);
```

Comment:
@GaneshSPatil I am not sure if this is the right place to schedule a material update. What happens if the configRepo is invalid or if the entity is not saved, we will still end up scheduling a material update.

Reply:
@maheshp -- I see what you're saying.
Currently, the EntityConfigUpdateCommand does not expose a post update method to perform post update validations/operations.
[This](https://github.com/gocd/gocd/issues/2780) issue talks about it.

`getPreprocessedEntityConfig` is the only method called post update. I am not sure whether the EntityConfigUpdateCommand is the right place to perform such operations.

How about moving it outside of the command and let configRepoService deal with it?

Reply:
Wondering if this can be done through a EntityConfigChangedListener?

Reply:
@maheshp -- That's a good option. I will give it a try.